woocommerce/woocommerce
# Migrate From Nx to Turborepo

### All Submissions:






### Changes proposed in this Pull Request:

This pull request replaces [Nx](http://nx.dev) with [Turborepo](http://turborepo.org). This pull request comes with a number of tooling and documentation updates to make sure that everything is clear. I plan to publish internal and external communication prior to and alongside this change to minimize the impact it may have on ongoing work.

Here is a mostly exhaustive list of the changes proposed in this pull request:

- Removal of Nx commands entirely. This means that commands like `pnpm nx composer-install` are no longer present in favor of running package manager commands directly. Note that in the case of Composer, each relevant package has a `"postinstall"` script that triggers a composer install automatically.
- Standardized Test Scripts
  - `test:e2e` -> `e2e`
  - `test:e2e-debug` -> `e2e:debug`
  - `test:e2e-dev` -> `e2e:dev`
  - `test:unit` -> `test
- Turborepo has been added as the executor of `build`, `lint`, `test`, and `e2e` commands. Use `pnpm exec turbo run {command} --filter={package}` to perform these actions. The output of the commands are cached where appropriate to minimize unnecessary executions. For example, `pnpm nx build woocommerce` is now `pnpm exec turbo run build --filter=woocommerce`.
- For other commands, use `pnpm`'s filter option to run them in the correct package. `pnpm changelog --filter=woocommerce` will run the `"changelog"` script from WooCommerce's `packge.json`.
- `woocommerce-admin` is now referred to as `@woocommerce/admin-library` since the filtering syntax uses the package name.

Closes #32655

### How to test the changes in this Pull Request:

1. Run `pnpm install`
2. Try out `build`, `lint`, `test`, and `e2e` commands. 
3. Make sure CI passes now that it was changed from Nx as well. 
### Other information:




### FOR PR REVIEWER ONLY:
